### PR TITLE
Add more SDR record types

### PR DESCRIPTION
**Pull request by datdenkikniet**
Copy of #10.

Closes #10

### CODE REVIEW

**Comment by datdenkikniet** [comment]
@richardstephens I've done some extra tinkering on this, and landed on just putting the relevant things in traits instead. I think it's good enough (for now, at least), since most of the relevant stuff is captured.

If you have time, I would appreciate it if you could put some eyes on it :)

File: src/storage/sdr/record/mod.rs [hunk 1]
```diff
@@ -627,24 +622,30 @@ impl<'a> TypeLengthRaw<'a> {
...
-            0b10 => SensorId::Ascii6BPacked(data.to_vec()),
-            0b11 => SensorId::Ascii8BAndLatin1(str.unwrap()),
+        let id = match type_code {
+            0b00 => SensorId::Unicode(
+                core::str::from_utf8(data)
+                    .map_err(|_| ParseError::InvalidSensorId)
+                    .map(str::to_string)?,
+            ),
+            0b01 => todo!(),
+            0b10 => todo!(),
```
Comment (richardstephens) [inline src/storage/sdr/record/mod.rs:643]: why leave these as todo's? 

We've never seen anything other than unicode, so I'd be okay with turning these into ParseError's, but I'd prefer these not panic.

File: src/storage/sdr/record/fru_device_locator.rs [hunk 2]
```diff
@@ -55,7 +55,8 @@ impl FruDeviceLocator {
             })
         };
 
-        let lun = LogicalUnit::try_from((record_data[2] >> 3) & 0b11).ok()?;
+        // NOTE(unwrap): `LogicalUnit::try_from` always succeeds for data masked with 0b11.
+        let lun = LogicalUnit::try_from((record_data[2] >> 3) & 0b11).unwrap();
```
Comment (richardstephens) [inline src/storage/sdr/record/fru_device_locator.rs:72]: How would you feel about moving the bitmask into a `LogicalUnit::from_low_bits(field:u8)` and doing the unwrap there?

**Comment by richardstephens** [comment]
Looks good.

Our fork has fallen a bit behind, so I'll make time over the next couple of days to bring it up to latest main + these changes.

File: src/storage/sdr/record/fru_device_locator.rs (hunk 2 above)
Comment (datdenkikniet) [reply src/storage/sdr/record/fru_device_locator.rs]: I'm a fan! Have added it

File: src/storage/sdr/record/mod.rs (hunk 1 above)
Comment (datdenkikniet) [reply src/storage/sdr/record/mod.rs]: Good point, wasn't thinking about this super hard. Replaced them with what we had before, and fixed up the display function to just print somewhat bogus data. We can correct it some other time.

**Pull request merged**